test/e2e/antreapolicy_test.go

```diff
@@ -3000,6 +3001,253 @@ func testACNPICMPSupport(t *testing.T, data *TestData) {
 	time.Sleep(networkPolicyDelay)
 }
 
+func testACNPIGMPQueryAllow(t *testing.T, data *TestData) {
+	nodeMulticastInterfaces, err := computeMulticastInterfaces(t, data)
+	if err != nil {
+		t.Fatalf("Error computing multicast interfaces: %v", err)
+	}
+	name := "test-acnp-igmp-query-allow"
+	mc := multicastTestcase{
+		name:            "testMulticastIGMPQueryAllow",
+		senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: false},
+		receiverConfigs: []multicastTestPodConfig{{1, false}},
+		port:            3457,
+		group:           net.ParseIP("224.3.4.13"),
+	}
+
+	testACNPIGMPQuery(t, data, name, mc, nodeMulticastInterfaces, crdv1alpha1.RuleActionAllow)
+}
+
+func testACNPIGMPQueryDrop(t *testing.T, data *TestData) {
+	nodeMulticastInterfaces, err := computeMulticastInterfaces(t, data)
+	if err != nil {
+		t.Fatalf("Error computing multicast interfaces: %v", err)
+	}
+	name := "test-acnp-igmp-query-drop"
+	mc := multicastTestcase{
+		name:            "testMulticastIGMPQueryDrop",
+		senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: false},
+		receiverConfigs: []multicastTestPodConfig{{1, false}},
+		port:            3457,
+		group:           net.ParseIP("224.3.4.14"),
+	}
+	testACNPIGMPQuery(t, data, name, mc, nodeMulticastInterfaces, crdv1alpha1.RuleActionDrop)
+}
+
+func testACNPIGMPQuery(t *testing.T, data *TestData, name string, mc multicastTestcase, nodeMulticastInterfaces map[int][]string, action crdv1alpha1.RuleAction) {
+	mcjoinWaitTimeout := defaultTimeout / time.Second
+	testNamespace := data.testNamespace
+	senderName, _, cleanupFunc := createAndWaitForPod(t, data, data.createMcJoinPodOnNode, "test-sender-", nodeName(mc.senderConfig.nodeIdx), testNamespace, mc.senderConfig.isHostNetwork)
+	defer cleanupFunc()
+	var wg sync.WaitGroup
+	receiverNames, cleanupFuncs := setupReceivers(t, data, mc, mcjoinWaitTimeout, &wg)
+	for _, cleanupFunc := range cleanupFuncs {
+		defer cleanupFunc()
+	}
+	// Wait 2 seconds(-w 2) before sending multicast traffic.
+	// It sends two multicast packets for every second(-f 500 means it takes 500 milliseconds for sending one packet).
+	sendMulticastCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -f 500 -o -p %d -s -t 3 -w 2 -W %d %s", mc.port, mcjoinWaitTimeout, mc.group.String())}
+	go func() {
+		data.RunCommandFromPod(testNamespace, senderName, mcjoinContainerName, sendMulticastCommand)
+	}()
+
+	tcpdumpName, _, cleanupFunc := createAndWaitForPod(t, data, data.createNetshootPodOnNode, "test-tcpdump-", nodeName(mc.receiverConfigs[0].nodeIdx), testNamespace, true)
+	defer cleanupFunc()
+
+	groupAddress := "224.0.0.1"
+	cmd, err := generateConnCheckCmd(t, data, 15, groupAddress, nodeName(mc.receiverConfigs[0].nodeIdx), receiverNames[0])
```

Comment:
Maybe rename it as "generatePacketCaptureCmd" ?

Reply:
done